src/dune_rules/scope.ml

```diff
@@ -277,7 +277,7 @@ module DB = struct
     Path.Source.Map.merge db_by_project_dir coq_db_by_project_dir
       ~f:(fun _dir project_and_db coq_db ->
         let project, db = Option.value_exn project_and_db in
-        let coq_db = Option.value_exn coq_db in
+        let coq_db = lazy (Option.value_exn coq_db) in
```

Comment:
This is happening too late, so this PR doesn't do anything. Lazy should be used in coq_db_by_project_dir

Reply:
In particular the place to be wrapped seems `             Coq_lib.DB.create_from_coqlib_stanzas
`

Reply:
How can I merge the source maps without forcing it tho?

Reply:
Because the maps are merged on the keys, here is only the value that is lazy.